builtin/credential/aws/path_role.go

```diff
@@ -241,6 +282,16 @@ func (b *backend) nonLockedAWSRole(s logical.Storage, roleName string) (*awsRole
 		}
 	}
 
+	// Check if there was no pre-existing AuthType set (from older versions)
```

Comment:
Also, code path not being safe is *not* an excuse for not saving the upgrade :-) We must persist the upgraded role.